aleph-client/src/rpc.rs

```diff
@@ -10,18 +11,94 @@ fn json_req(method: &str, params: Value, id: u32) -> Value {
     })
 }
 
-pub fn author_rotate_keys() -> Value {
+pub fn author_rotate_keys_json() -> Value {
     json_req("author_rotateKeys", Value::Null, 1)
 }
 
+fn state_query_storage_at_json(storage_keys: &Vec<StorageKey>) -> Value {
+    json_req(
+        "state_queryStorageAt",
+        Value::Array(vec![
+            Value::Array(
+                storage_keys
+                    .into_iter()
+                    .map(|storage_key| Value::String(hex::encode(storage_key)))
+                    .collect::<Vec<_>>(),
+            ),
+            Value::Null,
+        ]),
+        1,
+    )
+}
+
+fn parse_query_storage_at_result(
+    maybe_json_result: Option<String>,
+) -> Result<Vec<Option<String>>, String> {
+    match maybe_json_result {
+        None => Err(String::from("Returned result was null!")),
+        Some(result) => {
+            let mut storage_entries_in_hex_string = vec![];
+
+            let parsed_json_value: Value = serde_json::from_str(&result[..]).map_err(|_| {
+                String::from(&format!("Failed to parse result {:?} into JSON", result))
+            })?;
+            let storage_entries = match parsed_json_value[0]["changes"].as_array() {
```

Comment:
As mentioned in slack:
1) I would be careful with just taking `[0]` here without fully understanding what it means. The fact that this call outputs a vector has some meaning.
2) The proper way to parse this should be something like:
`let parsed: Vec<StorageChangeSet<Hash>> = serde_json::from_str(&result[..]).unwrap();` -- where `StorageChangeSet<Hash>` you can probably even copy from `substrate/client/rpc-api/src/state/mod.rs` or import from substrate

Reply:
Great comment, done.